src/HotChocolate/Core/src/Execution/Processing/VariableRewriter.cs

```diff
@@ -69,12 +69,23 @@ public static class VariableRewriter
 
         if (node.Fields.Count == 1)
         {
+            bool oneOf = type.Directives.Contains(WellKnownDirectives.OneOf);
```

Comment:
The validation actually guarantees that the value will not be null. We could remove this.